### PR TITLE
WIP: configure: Fix CXX expanded before required warning

### PR DESCRIPTION
**Pull request by pdxjohnny**
@tstruk The warning issue introduced in #1579 should be fixed

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -286,8 +288,7 @@ AS_CASE(["x$with_fuzzing"],
         [AC_MSG_ERROR([Bad value for --with-fuzzing])])
 AM_CONDITIONAL([ENABLE_FUZZING],[test "x$with_fuzzing" != "xnone"])
 AS_IF([test "x$with_fuzzing" != "xnone"],
-      [AC_PROG_CXX
-       AS_IF([test "x$enable_tcti_fuzzing" = xno],
```
Comment (williamcroberts) [inline configure.ac]: I was wondering about this. I pulled it in and didn't see the warning, weird.

**Comment by williamcroberts** [comment]
```
checking whether to build with code coverage support... no
checking that generated files are newer than configure... done
configure: error: conditional "am__fastdepCXX" was never defined.
Usually this means the macro was only invoked conditionally.
The command "./.ci/travis.run" exited with 1.
```

**Comment by pdxjohnny** [comment]
Sorry should have said, I haven't figured this out yet

**Comment by williamcroberts** [comment]
> Sorry should have said, I haven't figured this out yet

Should I revert your original patch till you get it sorted?

**Comment by pdxjohnny** [comment]
Sure ya that might be good, sorry

**Comment by williamcroberts** [comment]
> Sure ya that might be good, sorry

No worries, revert PR here:
https://github.com/tpm2-software/tpm2-tss/pull/1584

**Comment by diabonas** [comment]
Calling `AC_PROG_CXX` conditionally is hard, see e.g. the Automake mailing list post ["Can't call AC_PROG_CXX conditionally"](https://lists.gnu.org/archive/html/bug-automake/2002-11/msg00024.html). Instead I suggest simply calling `AC_PROG_CXX` after `AC_PROG_C` because that does not fail if no working C++ compiler is found (but sets it up correctly if one is available), see my PR #1588.

**Comment by pdxjohnny** [comment]
Awesome!!

**Pull request closed without merging**